src/programs/samples/samples_functional_testing.cpp

```diff
@@ -1,64 +1,59 @@
 //#include <wx/wx.h>
 
-#include "../../core/core_headers.h"
-#include "0_Simple/disk_io_image.cpp"
+// #include "common/samples_headers.h"
 #ifdef ENABLEGPU
-#include "1_GPU_comparison/cpu_vs_gpu.cpp"
+  #include "../../gpu/gpu_core_headers.h"
+#else
+  #include "../../core/core_headers.h"
 #endif
 
-#include "classes/TestFile.cpp"
-#include "classes/EmbeddedTestFile.cpp"
-#include "classes/NumericTestFile.cpp"
-#include "samples_functional_testing.h"
-// #define PrintResult(result)	PrintResultWorker(result, __LINE__);
-// #define FailTest {if (test_has_passed == true) PrintResultWorker(false,
-// __LINE__); test_has_passed = false;}//#include
-// "samples_functional_testing.hpp"
+// Helper classes for interacting with the test data.
+#include "common/common.h"
 
-// TODO //
-// TEST 3D's
-// ODD Sized IMAGES
-// NON SQUARE IMAGES
+// Sample functions
+#include "0_simple/simple.h"
+#include "1_cpu_gpu_comparison/cpu_gpu_comparison.h"
 
+// Test data
 #include "../console_test/hiv_image_80x80x1.cpp"
 #include "../console_test/hiv_images_shift_noise_80x80x10.cpp"
 #include "../console_test/sine_128x128x1.cpp"
 
-IMPLEMENT_APP(SamplesTestingApp);
+#include "samples_functional_testing.h"
+
+IMPLEMENT_APP(SamplesTestingApp)
 
-bool SamplesTestingApp::OnInit() {
+bool SamplesTestingApp::DoCalculation() 
+{
 
-  wxPrintf("Starting samples testing: \n\n");
-  
-  DoDiskIOImageTests(hiv_images_80x80x10_filename, temp_directory);
+  // This is returned and for automated testing allows you to only visualize the overall results,
+  // and if false, using github actions you can expand to see the full results.
+  bool all_tests_passed = true;
+
+  SamplesPrintTestStartMessage("Starting samples testing", true);
+
+  all_tests_passed = all_tests_passed && DoDiskIOImageTests(hiv_images_80x80x10_filename, temp_directory);
 
   #ifdef ENABLEGPU
-  DoGPUComplexResize(hiv_image_80x80x1_filename, temp_directory);
-    //DoCPUvsGPUResize(hiv_image_80x80x1_filename, temp_directory);
+    all_tests_passed = all_tests_passed && DoCPUvsGPUResize(hiv_image_80x80x1_filename, temp_directory);
   #else
     wxPrintf("GPU support disabled. skipping GPU tests.\n");
   #endif
 
-  wxPrintf("\nSamples testing done!\n\n");
-  return false;
+  SamplesPrintTestStartMessage("Samples testing done!", true);
+  ProgramSpecificCleanUp();
+  
+  return true;
 }
 
```

Comment:
Should this be `return all_tests_passed;`

Reply:
Oh shoot, you are right it should be. I'm going to be doing more work in the sample functions over the next few days, so I'll create a seperate issue for this and fix it then since the PR is otherwise good to go.